src/besselj.jl

```diff
@@ -255,6 +255,40 @@ function _besselj(nu::Integer, x::T) where T <: Union{Float32, Float64}
     end
 end
 
+function _besselj(nu::AbstractRange, x::T) where T
+    (nu[1] >= 0 && step(nu) == 1) || throw(ArgumentError("nu must be >= 0 with step(nu)=1"))
+    len = length(nu)
+    isone(len) && return [besselj(nu[1], x)]
+
+    out = zeros(T, len)
+    if nu[end] < x
+        out[1], out[2] = _besselj(nu[1], x), _besselj(nu[2], x)
+        return besselj_up_recurrence!(out, x, nu)
+    else
+        k = len
+        jn = zero(T)
+        while abs(jn) < floatmin(T)
+            if besselj_underflow_check(nu[k], x)
+                jn = zero(T)
+            else
+                jn = _besselj(nu[k], x)
+            end
+            out[k] = jn
+            k -= 1
+            k < 1 && break
+        end
+        if k > 1
+            out[k] = _besselj(nu[k], x)
+            out[1:k+1] = besselj_down_recurrence!(out[1:k+1], x, nu[1:k+1])
+            return out
+        else
+            return out
+        end
+    end
+end
```

Comment:
@oscardssmith Could you look at these lines? I think I'm creating unnecessary slices here. And for some reason this PR is also increasing allocations during broadcasting...
```julia
# MASTER
julia> @benchmark besselj.(1:200, 100.0)
BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  43.333 μs … 66.209 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     43.542 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   44.040 μs ±  1.517 μs  ┊ GC (mean ± σ):  0.00% ± 0.00%

  ▇█▅▃▂▂▁▁   ▁▄▄▃▂▁    ▁                                      ▂
  ██████████████████▇▇▆██▇███▇▇▆▆▆▅▅▆▇▆▅▃▅▁▄▁▅▄▄▆▄▁▄▄▃▅▆▄▅▅▅▄ █
  43.3 μs      Histogram: log(frequency) by time      51.1 μs <

 Memory estimate: 1.77 KiB, allocs estimate: 1.

# THIS PR
julia> @benchmark besselj.(1:200, 100.0)
BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  60.916 μs …  1.395 ms  ┊ GC (min … max): 0.00% … 95.40%
 Time  (median):     61.500 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   64.875 μs ± 33.526 μs  ┊ GC (mean ± σ):  0.97% ±  2.13%

  █▆▄▄▄▂▁                                                     ▁
  █████████▇▇▇▇▇▆▅▅▄▄▃▄▃▃▄▄▃▃▄▃▁▃▁▃▄▁▃▁▁▁▁▁▁▁▁▃▃▁▃▄▁▁▃▁▁▃▃▁▄▃ █
  60.9 μs      Histogram: log(frequency) by time       133 μs <

 Memory estimate: 25.31 KiB, allocs estimate: 1385.
```

Reply:
Though this PR significantly reduces this time.
```julia
# This PR no broadcast
julia> @benchmark besselj(1:200, 100.0)
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.379 μs … 75.442 μs  ┊ GC (min … max): 0.00% … 97.56%
 Time  (median):     1.462 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.555 μs ±  2.070 μs  ┊ GC (mean ± σ):  3.75% ±  2.76%

   ▃▃▅▇███▇▆▅▃▃▁                   ▁▂▂▂▂▁                    ▃
  ███████████████▆▆▃▁▃▁▁▁▃▄▃▁▁▁▁▅▆████████▇█▇▆▅▅▅▆▆▇██▇█▇▇▆▆ █
  1.38 μs      Histogram: log(frequency) by time     2.08 μs <

 Memory estimate: 3.66 KiB, allocs estimate: 10.
```

Reply:
`@views`?

Reply:
i.e
```if k > 1
            out[k] = _besselj(nu[k], x)
            @views besselj_down_recurrence!(out[1:k+1], x, nu[1:k+1])
            return out
```

Reply:
`@views` did remove an allocation. Something funny is going on as my old session with Revise is showing 1 allocation where new sessions are showing 150 allocations. Ill have to dig into this deeper.....